python/paddle/v2/fluid/framework.py

```diff
@@ -695,11 +702,22 @@ def var_recursive(self, name):
             return self.var(name)
         else:
             if self.idx == 0:
-                raise ValueError("var %s is not in block(%d) nor its parents." %
-                                 name, self.idx)
+                raise ValueError(
+                    "var {0} is not in block({1}) nor its parents.".format(
+                        name, self.idx))
             else:
-                parent_block = self.program.block(self.parent_idx)
-                return parent_block.var_recursive(name)
+                # DFS
```

Comment:
In issue #8367, the search is described as BFS. But the code here is DFS. Please confirm.

Reply:
Yes. I think the `BFS` should be the correct logic, however, DFS can be implemented easily. I will change the logic to `BFS` in following commits.